cmd/kyma/upgrade/cmd.go

```diff
@@ -0,0 +1,138 @@
+	cobraCmd.Flags().StringVarP(&o.TLSKey, "tlsKey", "", "", "TLS key for the domain used for upgrade. The key must be a base64-encoded value.")
+	cobraCmd.Flags().DurationVarP(&o.Timeout, "timeout", "", 1*time.Hour, "Time-out after which CLI stops watching the upgrade progress.")
+	cobraCmd.Flags().StringVarP(&o.Password, "password", "p", "", "Predefined cluster password.")
+	cobraCmd.Flags().StringArrayVarP(&o.OverrideConfigs, "override", "o", nil, "Path to a YAML file with parameters to override.")
+	cobraCmd.Flags().StringVarP(&o.ComponentsConfig, "components", "c", "", "Path to a YAML file with component list to override.")
+
+	return cobraCmd
+}
+
+//Run runs the command
+func (cmd *command) Run() error {
+	if cmd.opts.CI {
+		cmd.Factory.NonInteractive = true
+	}
+
+	var err error
+	if cmd.K8s, err = kube.NewFromConfigWithTimeout("", cmd.KubeconfigPath, cmd.opts.Timeout); err != nil {
+		return errors.Wrap(err, "Could not initialize the Kubernetes client. Make sure your kubeconfig is valid")
+	}
+
+	s := cmd.NewStep("Reading cluster info from ConfigMap")
+	clusterConfig, err := installation.GetClusterInfoFromConfigMap(cmd.K8s)
+	if err != nil {
+		s.Failure()
+		return err
+	}
+	s.Successf("Cluster info read")
+
+	installationInstance := cmd.configureInstallation(clusterConfig)
+	result, err := installationInstance.InstallKyma(upgradeAction)
+	if err != nil {
+		return err
+	}
+	if result == nil {
+		return nil
+	}
+
+	if clusterConfig.IsLocal {
+		s = cmd.NewStep("Adding domains to /etc/hosts")
+		err = hosts.AddDevDomainsToEtcHosts(s, clusterConfig, cmd.K8s, cmd.opts.Verbose, cmd.opts.Timeout, cmd.opts.Domain)
+		if err != nil {
+			s.Failure()
+			return err
+		}
+		s.Successf("Domains added")
+	}
+
+	err = cmd.printSummary(result)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cmd *command) configureInstallation(clusterConfig installation.ClusterInfo) *installation.Installation {
+	return &installation.Installation{
+		Options: &installation.Options{
+			NoWait:           cmd.opts.NoWait,
+			Verbose:          cmd.opts.Verbose,
+			CI:               cmd.opts.CI,
+			NonInteractive:   cmd.Factory.NonInteractive,
+			Timeout:          cmd.opts.Timeout,
+			KubeconfigPath:   cmd.opts.KubeconfigPath,
+			Domain:           cmd.opts.Domain,
+			TLSCert:          cmd.opts.TLSCert,
+			TLSKey:           cmd.opts.TLSKey,
+			Password:         cmd.opts.Password,
+			OverrideConfigs:  cmd.opts.OverrideConfigs,
+			ComponentsConfig: cmd.opts.ComponentsConfig,
+			IsLocal:          clusterConfig.IsLocal,
+			LocalCluster: &installation.LocalCluster{
+				IP:       clusterConfig.LocalIP,
+				Profile:  clusterConfig.Profile,
+				Provider: clusterConfig.Provider,
+				VMDriver: clusterConfig.LocalVMDriver,
+			},
+		},
+	}
+}
+
+func (cmd *command) printSummary(result *installation.Result) error {
+	nicePrint := nice.Nice{}
+	if cmd.Factory.NonInteractive {
+		nicePrint.NonInteractive = true
+	}
```

Comment:
I know this code was already like this, but an improvement could be:
```suggestion
	nicePrint := nice.Nice{
		NonInteractive: cmd.Factory.NonInteractive,
	}
```